### ISSUE TITLE
Cannot reproduce copy last mse on Kitti-Caltech dataset

### ISSUE THREAD

**Issue by liuem607** [coxlab/prednet#73]
Hi, we tried to calculate copy last testing MSE score on Kitti-Caltech datasets, but we got 0.0022 instead of 0.007 as reported in your paper. I'm really curious how did you get this number? Thanks,

**Comment by bill-lotter** [comment]
Hi sorry for the confusion, but the issue may be that Caltech and Kitti are actually two different datasets, where Kitti is used to train the model and then in the paper, Caltech is used to test, whereas here the evaluation script just uses the Kitti validation set. Retrieving the raw Caltech data relies on matlab scripts etc and is more complicated

**Comment by SaulZhang** [comment]
@liuem607 Did you normalize the intensity of the image to 0~1 during your mean square error calculation?And Ialso have another question，@bill-lotter，I see that there are two modes for calculating mean square error in your script(data_utils.py line 26), one is "all" and the another is "unique". Which one do you use when calculating the copy last frame? If it is convenient, I hope you can open source the code for calculating copy last frame mse and ssim.Very grateful for your help!

**Comment by SaulZhang** [comment]
The copy last result I reproduced in the Caltech Pedestrian testing set is that in “all” mode, MSE = 26.95×10^-3, SSIM = 0.704. In "unique" mode,MSE = 26.82×10^-3, SSIM = 0.705. But in the paper MSE = 7.95 × 10^−3, SSIM = 0.762.I don't know what's wrong with my.implementationBelow is my relevant code.When calculating mse and ssim, the intensity of the image is between 0 and 1.
`import numpy as np
import skimage
import skimage.measure

import torch
import torch.nn.functional as F
from torch.autograd import Variable
import numpy
from math import exp


def mse(img1,img2):
    img1 = img1.numpy()
    img2 = img2.numpy()
    mse = np.mean((img1 - img2)**2)
    return mse

def psnr(img1, img2):
    if len(img1.size()) == 3:
        img1 = img1.unsqueeze(0)
    if len(img2.size()) == 3:
        img2 = img2.unsqueeze(0)

    img1 = img1.numpy()
    img2 = img2.numpy()

    assert len(img1.shape) == 4 #(b,c,h,w) adapt to Pytorch
    
    mse = ((img1 - img2)**2).mean(1).mean(1).mean(1)
    PIXEL_MAX = 1.0
    PSNR = 20 * np.log10(PIXEL_MAX / np.sqrt(mse))
    return PSNR.mean(0)

#--------------ssim---------------
def gaussian(window_size, sigma):
    gauss = torch.Tensor([exp(-(x - window_size//2)**2/float(2*sigma**2)) for x in range(window_size)])
    return gauss/gauss.sum()

def create_window(window_size, channel):
    _1D_window = gaussian(window_size, 1.5).unsqueeze(1)
    _2D_window = _1D_window.mm(_1D_window.t()).float().unsqueeze(0).unsqueeze(0)
    window = Variable(_2D_window.expand(channel, 1, window_size, window_size).contiguous())
    return window

def _ssim(img1, img2, window, window_size, channel, size_average = True):
    mu1 = F.conv2d(img1, window, padding = window_size//2, groups = channel)
    mu2 = F.conv2d(img2, window, padding = window_size//2, groups = channel)

    mu1_sq = mu1.pow(2)
    mu2_sq = mu2.pow(2)
    mu1_mu2 = mu1*mu2

    sigma1_sq = F.conv2d(img1*img1, window, padding = window_size//2, groups = channel) - mu1_sq
    sigma2_sq = F.conv2d(img2*img2, window, padding = window_size//2, groups = channel) - mu2_sq
    sigma12 = F.conv2d(img1*img2, window, padding = window_size//2, groups = channel) - mu1_mu2

    C1 = 0.01**2
    C2 = 0.03**2

    ssim_map = ((2*mu1_mu2 + C1)*(2*sigma12 + C2))/((mu1_sq + mu2_sq + C1)*(sigma1_sq + sigma2_sq + C2))

    if size_average:
        return ssim_map.mean()
    else:
        return ssim_map.mean(1).mean(1).mean(1)

def ssim(img1, img2, window_size = 11, size_average = True):
    if len(img1.size()) == 3:
        img1 = img1.unsqueeze(0)
    if len(img2.size()) == 3:
        img2 = img2.unsqueeze(0)
    (_, channel, _, _) = img1.size()
    window = create_window(window_size, channel)
    
    if img1.is_cuda:
        window = window.cuda(img1.get_device())
    window = window.type_as(img1)
    
    return _ssim(img1, img2, window, window_size, channel, size_average)`
